src/Aggregation/GeohashGrid.php

```diff
@@ -46,6 +46,18 @@ public function setPrecision(int $precision): self
         return $this->setParam('precision', $precision);
     }
 
+    /**
+     * Set the precision for this aggregation in meters or kilometers.
+     *
+     * @param string $precision a string like 100m or 1km
+     *
+     * @return $this
+     */
+    public function setDistancePrecision(string $precision): self
```

Comment:
I would be in favor to use the previous setter and widen its type.
To simplify the migration path to multi arguments types available with PHP 8.0 I would add a check to ensure it's an int or a string in the setter and throw a \TypeError if the type is anything else than an int or string.

See for examples: https://github.com/ruflin/Elastica/blob/master/src/Query/Terms.php#L65 or https://github.com/ruflin/Elastica/blob/master/src/Index.php#L402

Changing the type of the method is a BC break, but in this case it's also a bug fix, so adding a changelog for documenting this BC should be fine. WDYT @ruflin ?

Reply:
that was my concern that it is BC and also bugfix 

I can make it one method  and check inside it if argument is int or string

Reply:
++ on treating it as a bug fix and use the previous one.

Reply:
ok, changed